### PR TITLE
Add idle expiry watcher

### PR DESCRIPTION
**Pull request by kimlisa**
resolves https://github.com/gravitational/webapps/issues/363

#### Description
Add a idle watcher inside `Authenticated` component

#### Related PR
get the web idle config: https://github.com/gravitational/teleport/pull/7839

### CODE REVIEW

File: packages/teleport/src/lib/idleTimer/idleTimer.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { throttle } from 'lodash';
+import Logger from 'shared/libs/logger';
+import session from 'teleport/services/session';
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply packages/teleport/src/lib/idleTimer/idleTimer.ts]: merged with `Authenticated`

File: packages/teleport/src/lib/idleTimer/idleTimer.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+const logger = Logger.create('lib/idleTimer');
+
+// IDLE_EXPIRY is the local storage key to store idle expiry.
+const IDLE_EXPIRY = 'grv_teleport_idle_expiry';
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply packages/teleport/src/lib/idleTimer/idleTimer.ts]: revised

File: packages/teleport/src/components/Authenticated/Authenticated.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+};
+
+export default Authenticated;
+
+function startActivityChecker(inactivityTtl = 0) {
+  // adjustedInactivityTtl slightly improves accuracy of inactivity time.
+  // This will at most cause user to log out ACTIVITY_CHECKER_INTERVAL_MS early.
+  // NOTE: Because of browser js throttling on inactive tabs, expiry timeout may
+  // still be extended up to over a minute.
+  const adjustedInactivityTtl = inactivityTtl - ACTIVITY_CHECKER_INTERVAL_MS;
```
Comment (alex-kovoy) [inline packages/teleport/src/components/Authenticated/Authenticated.tsx:61]: `adjustedInactivityTtl` -> `ttl`

File: packages/teleport/src/components/Authenticated/Authenticated.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    clearInterval(intervalId);
+    window.removeEventListener('keydown', throttled);
+    window.removeEventListener('pointermove', throttled);
+    window.removeEventListener('pointerdown', throttled);
+  }
+
+  return stop;
+}
+
+function checkInactivityTimeout(inactivityTtl = 0) {
```
Comment (alex-kovoy) [inline packages/teleport/src/components/Authenticated/Authenticated.tsx:101]: `inactivityTtl` -> `ttl` (as function name sets a context)

**Review by alex-kovoy — APPROVED** [review]
with minor renames

File: packages/teleport/src/components/Authenticated/Authenticated.tsx (hunk 3 above)
Comment (kimlisa) [reply packages/teleport/src/components/Authenticated/Authenticated.tsx]: renamed

File: packages/teleport/src/components/Authenticated/Authenticated.tsx (hunk 4 above)
Comment (kimlisa) [reply packages/teleport/src/components/Authenticated/Authenticated.tsx]: renamed

**Review by alex-kovoy — APPROVED** [review]

**Pull request merged**